bolt/lib/Core/DebugData.cpp

```diff
@@ -906,6 +906,7 @@ void DebugStrOffsetsWriter::finalizeSection(DWARFUnit &Unit,
   }
 
   StrOffsetSectionWasModified = false;
+  DebugStrOffsetFinalized[Unit.getOffset()] = true;
```

Comment:
see if this can be part of assert with .insert and checking output bool for if it's a new entry.
This way if asserts are disabled we don't have to insert anything in this.
You might need [[maybe_unused]] on definition so that clang doesn't report warnings if it not being used when asserts are disabled.